File: boveda-digital-api/src/main/java/com/example/boveda_digital_api/services/ErrorLogService.java
package com.example.boveda_digital_api.services;

import com.example.boveda_digital_api.dao.ErrorLogRepository;
import com.example.boveda_digital_api.entity.ErrorLogEntity;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class ErrorLogService {

    private final ErrorLogRepository errorLogRepository;

    @Autowired
    public ErrorLogService(ErrorLogRepository errorLogRepository) {
        this.errorLogRepository = errorLogRepository;
    }

    public void saveError(String module, String publicMessage, String internalReason, String details) {
        try {
            ErrorLogEntity errorLog = new ErrorLogEntity();
            errorLog.setModule(module);
            errorLog.setPublicMessage(publicMessage);
            errorLog.setInternalReason(internalReason);
            errorLog.setDetails(details);

            errorLogRepository.save(errorLog);
        } catch (Exception e) {
            // No lanzar otra excepción para no romper el flujo principal
            System.err.println("No se pudo guardar el error en la base: " + e.getMessage());
        }
    }
}

File: boveda-digital-frontend/src/services/errorLogService.js
import axios from "axios";

const API_URL = "http://localhost:8080/api/error-logs";

export async function sendErrorLog({ module, publicMessage, internalReason, details }) {
  try {
    await axios.post(API_URL, {
      module,
      public_message: publicMessage,
      internal_reason: internalReason,
      details
    });
  } catch (e) {
    // No romper el flujo principal si falla el guardado del log
    console.warn("No se pudo guardar el error en backend");
  }
}